File: Algorithms/cpp/102 - Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        // var
        vector<int> merge;
        vector<vector<int>> result;
        
        if (!root)
            return result;
        
        // init
        result.push_back(vector<int>(1, root->val));
        
        // next level
        vector<vector<int>> left = levelOrder(root->left);;
        vector<vector<int>> right = levelOrder(root->right);;
        
        // merge subtree
        for (int i = 0; i < max(left.size(), right.size()); i++) {
            if (left.size() > i)
                merge.insert(merge.end(), left[i].begin(), left[i].end());
            if (right.size() > i)
                merge.insert(merge.end(), right[i].begin(), right[i].end());
            result.push_back(merge);
            merge.clear();
        }
        return result;
    }
};

File: python/102 - Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        # init
        level = []
        stack = [root]
        
        while len(stack):
            # new 
            _stack = []
            _level = []
            
            while len(stack):
                cur = stack.pop(0)
                
                if cur == None:
                    continue
                
                if cur.left:
                    _stack.append(cur.left)
                if cur.right:
                    _stack.append(cur.right)
                _level.append(cur.val)
                
            # cp
            for i in _stack:
                stack.append(i)

            if _level != []:
                level.append(_level)
            
        return level    